src/target/arc_ocd.c

```diff
@@ -99,6 +99,19 @@ int arc_ocd_assert_reset(struct target *target)
 
 		target_handle_event(target, TARGET_EVENT_RESET_ASSERT);
 		register_cache_invalidate(arc32->core_cache);
+		/*
+			An ARC target might be in halt state after reset, so
+			if script requested processor to resume, then it must
+			be manually started to ensure that this request
+			is satisfied.
+		*/
+		if (target->state == TARGET_HALTED && !target->reset_halt) {
```

Comment:
Missing `LOG_DEBUG` entry: https://github.com/foss-for-synopsys-dwc-arc-processors/openocd/pull/25#discussion_r188645892